catalog/spatial/csw/spatial-csw-endpoint/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/transformer/CswRecordMapperFilterVisitor.java

```diff
@@ -452,34 +406,9 @@ public Object visit(PropertyIsLessThanOrEqualTo filter, Object extraData) {
         Literal literal = getFactory(extraData).literal(orig);
         return getFactory(extraData).before(other, literal);
       }
-    } else {
-      AttributeType type =
-          attributeTypes.get(((PropertyName) filter.getExpression1()).getPropertyName());
-
-      LiteralExpressionImpl typedExpression = (LiteralExpressionImpl) filter.getExpression2();
-      setExpressionType(type, typedExpression);
-
-      expr2 = visit((Expression) typedExpression, expr1);
     }
-    return getFactory(extraData).lessOrEqual(expr1, expr2);
-  }
 
-  private void setExpressionType(AttributeType type, LiteralExpressionImpl typedExpression) {
-    if (type != null && typedExpression != null) {
-      if (type.getBinding() == Short.class) {
-        typedExpression.setValue(Short.valueOf((String) typedExpression.getValue()));
-      } else if (type.getBinding() == Integer.class) {
-        typedExpression.setValue(Integer.valueOf((String) typedExpression.getValue()));
-      } else if (type.getBinding() == Long.class) {
-        typedExpression.setValue(Long.valueOf((String) typedExpression.getValue()));
-      } else if (type.getBinding() == Float.class) {
-        typedExpression.setValue(Float.valueOf((String) typedExpression.getValue()));
-      } else if (type.getBinding() == Double.class) {
-        typedExpression.setValue(Double.valueOf((String) typedExpression.getValue()));
-      } else if (type.getBinding() == Boolean.class) {
-        typedExpression.setValue(Boolean.valueOf((String) typedExpression.getValue()));
-      }
-    }
```

Comment:
It's not overly clear from just looking at the code, but every time we visit a LiteralExpression we end up on line 424 which does the exact same data conversion at this method that I've removed.